mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDaoImpl.java

```diff
@@ -451,4 +460,64 @@ private String buildOrderByClause(List<SortField> sortFields) {
       .collect(Collectors.joining(", ", "ORDER BY ", EMPTY));
   }
 
+  @Override
+  public Future<RowSet<Row>> hardDeleteJobExecutions(String tenantName, long diffNumberOfDays) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    Promise<SQLConnection> connection = Promise.promise();
+    PostgresClient postgresClient = pgClientFactory.createInstance(tenantName);
+    String rollbackMessage = "Rollback transaction. Error during jobExecution deletion process.";
+    Future.succeededFuture()
```

Comment:
nitpick: thinking you could just start the async chain with `postgresClient.startTx` rather than `succeededFuture`. You might have a reason for starting from a `succeededFuture`.

Reply:
This method has been refactored